skimage/color/colorconv.py

```diff
@@ -1516,10 +1526,12 @@ def combine_stains(stains, conv_matrix):
     from ..exposure import rescale_intensity
 
     stains = _prepare_colorarray(stains)
-    logrgb2 = -np.reshape(stains, (-1, 3)) @ conv_matrix
-    rgb2 = np.power(10, logrgb2)
-    return rescale_intensity(np.reshape(rgb2 - 2, stains.shape),
-                             in_range=(-1, 1))
+    # log_adjust here is used to compensate the sum within separate_stains().
+    log_adjust = -np.log(1E-6)
+    log_rgb = -(stains * log_adjust) @ conv_matrix
+    rgb = np.power(np.e, log_rgb)
+
+    return rescale_intensity(rgb, in_range=(0, 1))
```

Comment:
Just like in the conversion in the other direction, I would get rid of the `rescale_intensity`. We do not want scaling that depends on image content, because it makes it impossible to compare results across images.

This min-max rescaling will also make the round-trip test fail unless the input happens to span the full [0,1] range. I suggest you add a test with data that does not span the full range to verify that there is no image-content-dependent stuff going on.

Reply:
You got it. Made some tests, it seems OK to leave it out.

Reply:
Nope... actually it will break our tests :slightly_smiling_face: it seems to a float between `(0, 1)` coming back from `combine_stains()`. I am re-running tests locally to check if this is the case.